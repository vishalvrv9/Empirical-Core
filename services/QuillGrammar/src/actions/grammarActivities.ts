import { push } from 'react-router-redux';
import rootRef from '../firebase';
import { ActionTypes } from './actionTypes'
const activitiesRef = rootRef.child('grammarActivities')
import { GrammarActivities, GrammarActivity } from '../interfaces/grammarActivities'

export const startListeningToActivities = () => {
  return (dispatch: Function) => {
    activitiesRef.on('value', (snapshot: any) => {
      const activities: GrammarActivities[] = snapshot.val()
      if (activities) {
        dispatch({ type: ActionTypes.RECEIVE_GRAMMAR_ACTIVITIES_DATA, data: activities, });
      } else {
        dispatch({ type: ActionTypes.NO_GRAMMAR_ACTIVITIES_FOUND })
      }
    });

  }
}

export const getActivity = (activityUID: string) => {
  return (dispatch: Function) => {
    activitiesRef.child(activityUID).once('value', (snapshot: any) => {
      const activity: GrammarActivity = snapshot.val()
      if (activity) {
        dispatch({ type: ActionTypes.RECEIVE_GRAMMAR_ACTIVITY_DATA, data: activity, });
      } else {
        dispatch({ type: ActionTypes.NO_GRAMMAR_ACTIVITY_FOUND })
      }
    });

  }
}

export const toggleNewLessonModal = () => {
  return { type: ActionTypes.TOGGLE_NEW_LESSON_MODAL, };
}

export const submitNewLesson = (content: GrammarActivity) => {
  const cleanedContent = _.pickBy(content)
  return (dispatch: Function) => {
    dispatch({ type: ActionTypes.AWAIT_NEW_LESSON_RESPONSE, });
    const newRef = activitiesRef.push(cleanedContent, (error: string) => {
      dispatch({ type: ActionTypes.RECEIVE_NEW_LESSON_RESPONSE, });
      if (error) {
        dispatch({ type: ActionTypes.DISPLAY_ERROR, error: `Submission failed! ${error}`, });
      } else {
        dispatch({ type: ActionTypes.DISPLAY_MESSAGE, message: 'Submission successfully saved!', });
        const action = push(`/admin/lessons/${newRef.key}`);
        dispatch(action);
      }
    });
  };
}

export const startLessonEdit = (cid: string) => {
  return { type: ActionTypes.START_LESSON_EDIT, cid, };
}

export const cancelLessonEdit = (cid: string) => {
  return { type: ActionTypes.FINISH_LESSON_EDIT, cid, };
}

export const submitLessonEdit = (cid: string, content: GrammarActivity) => {
  return (dispatch: Function) => {
    dispatch({ type: ActionTypes.SUBMIT_LESSON_EDIT, cid, });
    const cleanedContent = _.pickBy(content)
    activitiesRef.child(cid).set(cleanedContent, (error: string) => {
      dispatch({ type: ActionTypes.FINISH_LESSON_EDIT, cid, });
      if (error) {
        dispatch({ type: ActionTypes.DISPLAY_ERROR, error: `Update failed! ${error}`, });
      } else {
        dispatch({ type: ActionTypes.DISPLAY_MESSAGE, message: 'Update successfully saved!', });
      }
    });
  };
}

export const deleteLesson = (cid: string) => {
  return (dispatch: Function) => {
    dispatch({ type: ActionTypes.SUBMIT_LESSON_EDIT, cid, });
    activitiesRef.child(cid).remove((error: string) => {
      dispatch({ type: ActionTypes.FINISH_LESSON_EDIT, cid, });
      if (error) {
        dispatch({ type: ActionTypes.DISPLAY_ERROR, error: `Deletion failed! ${error}`, });
      } else {
        dispatch({ type: ActionTypes.DISPLAY_MESSAGE, message: 'Lesson successfully deleted!', });
        const action = push(`/admin/lessons`)
        dispatch(action)
      }
    });
  };
}
