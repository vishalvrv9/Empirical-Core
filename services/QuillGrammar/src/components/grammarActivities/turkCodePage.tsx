import * as React from 'react'

const TurkCodePage = () => (<div style={{fontSize: '24px', display: 'flex', justifyContent: 'center', marginTop: '100px'}}>
  <h1>Thank you for completing the Quill Grammar activity.</h1>
  <p>
    Your unique code is ij3982rmf9.
  </p>
</div>)

export default TurkCodePage
