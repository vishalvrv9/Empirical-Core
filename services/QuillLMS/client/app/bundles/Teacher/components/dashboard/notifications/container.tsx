import React from 'react';
import { ApolloProvider } from "react-apollo";
import client from '../../../../../modules/apollo';
import { Query } from "react-apollo";
import gql from "graphql-tag";
import NotificationsCard from './notification_feed';

export interface Notification {
  id: number
  text: string
  href: string|null
}

const notificationQuery:string = `
  {
    currentUser {
      notifications {
        id
        text
        href
      }
    }
  }
`

export default () => (
  <ApolloProvider client={client}>
    <Query
        query={gql(notificationQuery)}
      >
      {({ loading, error, data }) => {
        if (loading) return (null);
        if (error) return <p>Error :(</p>;
        const notifications:Notification[] = data.currentUser.notifications;
        return (
          <NotificationsCard notifications={notifications}/>
        )
      }}
    </Query>
  </ApolloProvider>
);