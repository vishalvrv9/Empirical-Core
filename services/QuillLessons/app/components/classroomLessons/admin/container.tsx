import React, { Component } from 'react';
import { connect } from 'react-redux';
import rootRef, { firebase } from '../../../libs/firebase';
import _ from 'lodash'
import {
  addLesson
} from '../../../actions/classroomLesson'
import {
  startListeningToEditionMetadata
} from '../../../actions/customize'
import {
  listenForClassroomLessons,
  listenForClassroomLessonReviews
} from '../../../actions/classroomLesson';

class AdminClassLessonsContainer extends Component<any, any> {
  constructor(props) {
    super(props);

    this.props.dispatch(listenForClassroomLessons());
    this.props.dispatch(listenForClassroomLessonReviews())
    this.props.dispatch(startListeningToEditionMetadata())
  }

  render() {
    return <div>{this.props.children}</div>
  }

}

function select(props) {
  return {
    classroomLessons: props.classroomLessons,
    classroomLessonsReviews: props.classroomLessonsReviews,
    customize: props.customize
  };
}

function mergeProps(stateProps: Object, dispatchProps: Object, ownProps: Object) {
  return {...ownProps, ...stateProps, ...dispatchProps}
}

export default connect(select, dispatch => ({dispatch}), mergeProps)(AdminClassLessonsContainer);
