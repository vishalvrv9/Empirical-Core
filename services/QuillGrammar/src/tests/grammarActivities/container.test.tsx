import * as React from "react";
import {shallow} from "enzyme";
import { PlayGrammarContainer } from "../../components/grammarActivities/container";
import { grammarActivities, session, conceptsFeedback } from './data'

process.env.EMPIRICAL_BASE_URL = 'https://staging.quill.org'
process.env.QUILL_CMS = 'https://cms.quill.org'

describe("<PlayGrammarContainer />", () => {
  const wrapper = shallow(<PlayGrammarContainer
    dispatch={() => {}}
    grammarActivities={grammarActivities}
    session={session}
    conceptsFeedback={conceptsFeedback}
   />)

    it("should render", () => {
      expect(wrapper).toMatchSnapshot();
    });

});
