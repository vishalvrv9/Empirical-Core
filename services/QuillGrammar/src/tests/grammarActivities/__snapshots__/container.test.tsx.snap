// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayGrammarContainer /> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayGrammarContainer
    conceptsFeedback={
      Object {
        "data": Object {},
        "hasreceiveddata": false,
        "newConceptModalOpen": false,
        "states": Object {},
        "submittingnew": false,
      }
    }
    dispatch={[Function]}
    grammarActivities={
      Object {
        "currentActivity": Object {
          "concepts": Object {
            "1ohLyApTz7lZ3JszrA98Xg": Object {
              "quantity": 9,
            },
          },
          "description": "Write nine sentences using parallel structure.",
          "title": "Parallel Structure",
        },
        "hasreceiveddata": true,
      }
    }
    session={
      Object {
        "answeredQuestions": Array [],
        "currentQuestion": Object {
          "answers": Array [
            Object {
              "text": "The team stretched their legs, {practiced} their drills, and then rehearsed their set plays.",
            },
          ],
          "attempts": Array [],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence so that it has proper parallel structure.",
          "prompt": "The team stretched their legs, practice their drills, and then rehearsed their set plays.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-Jzw0qjO5owyFPUAwDGx",
        },
        "hasreceiveddata": true,
        "unansweredQuestions": Array [
          Object {
            "answers": Array [
              Object {
                "text": "I prefer {reading the book than watching the movie adaptation}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "I prefer <u>reading the book than to watch the movie adaptation</u>.",
            "rule_description": "<b>Correct: </b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect: </b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9xwXu4o93MS_0NjR",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The crossing guard {was holding out his stop sign and ushering} students across the street.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "The crossing guard <u>was holding out his stop sign and ushered</u> students across the street.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9y8hvHU0-OXp1Buq",
          },
          Object {
            "answers": Array [
              Object {
                "text": "On my vacation to Universal Studios, I will {ride roller coasters and play carnival games}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "On my vacation to Universal Studios, I will <u>ride roller coasters and playing carnival games</u>.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9yDOc5nKBI-kEeHI",
          },
          Object {
            "answers": Array [
              Object {
                "text": "Shannon loves to play the flute, march in the band, and {watch} the football games.",
              },
              Object {
                "text": "Shannon loves {playing the flute, marching in the band, and} watching the football games.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "Shannon loves to play the flute, march in the band, and watching the football games.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfz8_NHXJ2hrQZRyxU",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The boss wanted to hire people who arrived on time, {smiled} at customers, and didn't take too many breaks.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "The boss wanted to hire people who arrived on time, would smile at customers, and didn't take too many breaks.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfxB8z_R1vkm8-WuWY",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The Rocky Mountains are a great place {for skiing, hiking, and mountain biking}.",
              },
              Object {
                "text": "The Rocky Mountains are a great place {to ski, hike, and to mountain bike}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "The Rocky Mountains are a great place <u>for skiing, hiking, and to mountain bike</u>.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9yAQ_YyWuT9x4e17",
          },
          Object {
            "answers": Array [
              Object {
                "text": "During a conversation, talking is just as important as {listening}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "During a conversation, talking is just as important as to listen.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfwKGn8uN5BGLgXEDa",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The teacher graded the students' work quickly, accurately, and {thoroughly.}",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "The teacher graded the students' work quickly, accurately, and thorough.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfwjD8pcldAXkHjxA8",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activity": Object {
        "concepts": Object {
          "1ohLyApTz7lZ3JszrA98Xg": Object {
            "quantity": 9,
          },
        },
        "description": "Write nine sentences using parallel structure.",
        "title": "Parallel Structure",
      },
      "answeredQuestions": Array [],
      "checkAnswer": [Function],
      "concepts": undefined,
      "conceptsFeedback": Object {
        "data": Object {},
        "hasreceiveddata": false,
        "newConceptModalOpen": false,
        "states": Object {},
        "submittingnew": false,
      },
      "currentQuestion": Object {
        "answers": Array [
          Object {
            "text": "The team stretched their legs, {practiced} their drills, and then rehearsed their set plays.",
          },
        ],
        "attempts": Array [],
        "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
        "instructions": "Rewrite the sentence so that it has proper parallel structure.",
        "prompt": "The team stretched their legs, practice their drills, and then rehearsed their set plays.",
        "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
        "uid": "-Jzw0qjO5owyFPUAwDGx",
      },
      "goToNextQuestion": [Function],
      "unansweredQuestions": Array [
        Object {
          "answers": Array [
            Object {
              "text": "I prefer {reading the book than watching the movie adaptation}.",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence with the correct parallel structure.",
          "prompt": "I prefer <u>reading the book than to watch the movie adaptation</u>.",
          "rule_description": "<b>Correct: </b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect: </b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAg9xwXu4o93MS_0NjR",
        },
        Object {
          "answers": Array [
            Object {
              "text": "The crossing guard {was holding out his stop sign and ushering} students across the street.",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence with the correct parallel structure.",
          "prompt": "The crossing guard <u>was holding out his stop sign and ushered</u> students across the street.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAg9y8hvHU0-OXp1Buq",
        },
        Object {
          "answers": Array [
            Object {
              "text": "On my vacation to Universal Studios, I will {ride roller coasters and play carnival games}.",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence with the correct parallel structure.",
          "prompt": "On my vacation to Universal Studios, I will <u>ride roller coasters and playing carnival games</u>.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAg9yDOc5nKBI-kEeHI",
        },
        Object {
          "answers": Array [
            Object {
              "text": "Shannon loves to play the flute, march in the band, and {watch} the football games.",
            },
            Object {
              "text": "Shannon loves {playing the flute, marching in the band, and} watching the football games.",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence so that it has proper parallel structure.",
          "prompt": "Shannon loves to play the flute, march in the band, and watching the football games.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAfz8_NHXJ2hrQZRyxU",
        },
        Object {
          "answers": Array [
            Object {
              "text": "The boss wanted to hire people who arrived on time, {smiled} at customers, and didn't take too many breaks.",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence so that it has proper parallel structure.",
          "prompt": "The boss wanted to hire people who arrived on time, would smile at customers, and didn't take too many breaks.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAfxB8z_R1vkm8-WuWY",
        },
        Object {
          "answers": Array [
            Object {
              "text": "The Rocky Mountains are a great place {for skiing, hiking, and mountain biking}.",
            },
            Object {
              "text": "The Rocky Mountains are a great place {to ski, hike, and to mountain bike}.",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence with the correct parallel structure.",
          "prompt": "The Rocky Mountains are a great place <u>for skiing, hiking, and to mountain bike</u>.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAg9yAQ_YyWuT9x4e17",
        },
        Object {
          "answers": Array [
            Object {
              "text": "During a conversation, talking is just as important as {listening}.",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence so that it has proper parallel structure.",
          "prompt": "During a conversation, talking is just as important as to listen.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAfwKGn8uN5BGLgXEDa",
        },
        Object {
          "answers": Array [
            Object {
              "text": "The teacher graded the students' work quickly, accurately, and {thoroughly.}",
            },
          ],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence so that it has proper parallel structure.",
          "prompt": "The teacher graded the students' work quickly, accurately, and thorough.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-KAfwjD8pcldAXkHjxA8",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activity": Object {
          "concepts": Object {
            "1ohLyApTz7lZ3JszrA98Xg": Object {
              "quantity": 9,
            },
          },
          "description": "Write nine sentences using parallel structure.",
          "title": "Parallel Structure",
        },
        "answeredQuestions": Array [],
        "checkAnswer": [Function],
        "concepts": undefined,
        "conceptsFeedback": Object {
          "data": Object {},
          "hasreceiveddata": false,
          "newConceptModalOpen": false,
          "states": Object {},
          "submittingnew": false,
        },
        "currentQuestion": Object {
          "answers": Array [
            Object {
              "text": "The team stretched their legs, {practiced} their drills, and then rehearsed their set plays.",
            },
          ],
          "attempts": Array [],
          "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
          "instructions": "Rewrite the sentence so that it has proper parallel structure.",
          "prompt": "The team stretched their legs, practice their drills, and then rehearsed their set plays.",
          "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
          "uid": "-Jzw0qjO5owyFPUAwDGx",
        },
        "goToNextQuestion": [Function],
        "unansweredQuestions": Array [
          Object {
            "answers": Array [
              Object {
                "text": "I prefer {reading the book than watching the movie adaptation}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "I prefer <u>reading the book than to watch the movie adaptation</u>.",
            "rule_description": "<b>Correct: </b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect: </b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9xwXu4o93MS_0NjR",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The crossing guard {was holding out his stop sign and ushering} students across the street.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "The crossing guard <u>was holding out his stop sign and ushered</u> students across the street.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9y8hvHU0-OXp1Buq",
          },
          Object {
            "answers": Array [
              Object {
                "text": "On my vacation to Universal Studios, I will {ride roller coasters and play carnival games}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "On my vacation to Universal Studios, I will <u>ride roller coasters and playing carnival games</u>.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9yDOc5nKBI-kEeHI",
          },
          Object {
            "answers": Array [
              Object {
                "text": "Shannon loves to play the flute, march in the band, and {watch} the football games.",
              },
              Object {
                "text": "Shannon loves {playing the flute, marching in the band, and} watching the football games.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "Shannon loves to play the flute, march in the band, and watching the football games.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfz8_NHXJ2hrQZRyxU",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The boss wanted to hire people who arrived on time, {smiled} at customers, and didn't take too many breaks.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "The boss wanted to hire people who arrived on time, would smile at customers, and didn't take too many breaks.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfxB8z_R1vkm8-WuWY",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The Rocky Mountains are a great place {for skiing, hiking, and mountain biking}.",
              },
              Object {
                "text": "The Rocky Mountains are a great place {to ski, hike, and to mountain bike}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence with the correct parallel structure.",
            "prompt": "The Rocky Mountains are a great place <u>for skiing, hiking, and to mountain bike</u>.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAg9yAQ_YyWuT9x4e17",
          },
          Object {
            "answers": Array [
              Object {
                "text": "During a conversation, talking is just as important as {listening}.",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "During a conversation, talking is just as important as to listen.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfwKGn8uN5BGLgXEDa",
          },
          Object {
            "answers": Array [
              Object {
                "text": "The teacher graded the students' work quickly, accurately, and {thoroughly.}",
              },
            ],
            "concept_uid": "1ohLyApTz7lZ3JszrA98Xg",
            "instructions": "Rewrite the sentence so that it has proper parallel structure.",
            "prompt": "The teacher graded the students' work quickly, accurately, and thorough.",
            "rule_description": "<b>Correct:</b> I laughed, I cried, and I shouted during the performance.
<b>Incorrect:</b> I laughed, I was crying, and I shout during the performance.
Sentences flow easily when all the actions in a list have the same ending. This means that each and every action in a list stays in the past, present, or future tense.",
            "uid": "-KAfwjD8pcldAXkHjxA8",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
